unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -59,257 +53,89 @@ export function QuerySuggestPreviewTest() {
-      triggerQuerySuggestHover(suggestion);
+    function triggerPopulateSearchResultPreviewsAndPassTime(suggestion: string = 'test', fakeResults?: IQueryResults) {
+      const query = triggerPopulateSearchResultPreviews(suggestion);
       jasmine.clock().tick(test.cmp.options.executeQueryDelay);
-    }
-
-    function buildSuggestion() {
-      container = $$(document.createElement('div'));
-      suggestionContainer = $$(document.createElement('div'));
-      suggestion = $$(document.createElement('div'));
-      elementInsideSuggestion = $$(document.createElement('div'));
-
-      suggestion.el.appendChild(elementInsideSuggestion.el);
-      suggestionContainer.el.appendChild(suggestion.el);
-      container.el.appendChild(suggestionContainer.el);
-    }
-
-    function setupSuggestionsManager() {
-      buildSuggestion();
-      const inputManager = new InputManager(document.createElement('div'), () => {}, {} as MagicBoxInstance);
-
-      const container = document.createElement('div');
-      suggestionManager = new SuggestionsManager(suggestionContainer.el, container, inputManager, {});
-    }
-
-    function setupRenderPreview() {
-      test.env.root.appendChild(suggestionContainer.el);
-      (test.cmp.options.resultTemplate.instantiateToElement as jasmine.Spy).and.returnValue(Promise.resolve($$('div').el));
-    }
-
-    function setupSuggestion(suggestions: Suggestion[] = [{ text: 'test' }]) {
-      setupSuggestionsManager();
-      setupRenderPreview();
-      suggestionManager.updateSuggestions(suggestions);
-    }
-
-    function waitXms(ms: number) {
-      return Promise.resolve(
-        setTimeout(() => {
-          return;
-        }, ms)
-      );
+      return query;
     }
 
     beforeEach(() => {
       testEnv = new Mock.MockEnvironmentBuilder();
       omniboxAnalytics = this.initOmniboxAnalyticsMock(omniboxAnalytics);
       testEnv.searchInterface.getOmniboxAnalytics = jasmine.createSpy('omniboxAnalytics').and.returnValue(omniboxAnalytics) as any;
+      jasmine.clock().install();
     });
 
-    describe('expose options', () => {
-      beforeEach(() => {
-        jasmine.clock().install();
-      });
-
-      afterEach(() => {
-        jasmine.clock().uninstall();
-      });
+    afterEach(() => {
+      jasmine.clock().uninstall();
+    });
 
-      it('numberOfPreviewResults set the number of results to query', () => {
+    describe('expose options', () => {
+      it('numberOfPreviewResults set the number of results to query', async done => {
         const numberOfPreviewResults = 5;
         setupQuerySuggestPreview({ numberOfPreviewResults });
-        setupSuggestion();
-        triggerQuerySuggestHoverAndPassTime();
+        await triggerPopulateSearchResultPreviewsAndPassTime();
         expect(test.cmp.queryController.getLastQuery().numberOfResults).toBe(numberOfPreviewResults);
+        done();
```

Comment:
Since results are now part of a promise, an `await` is necessary to obtain them, even though `jasmine.clock().tick()` resolves the promise instantly.

Reply:
👍